src/main/java/edu/pdx/cs/Client.java

```diff
@@ -37,6 +38,72 @@ void setLogfile(Path logpath) {
         this.logpath = logpath;
     }
 
+    /**
+     * Lists directories w/in the current working directory on remote server
+     * @return an array of <code>FTPFile</code> objects
+     */
+    protected FTPFile[] listRemoteDirectories() throws IOException {
+        String path = ftp.printWorkingDirectory();
+        return ftp.listDirectories(path);
+    }
+
+    /**
+     * Lists files w/in the current directory on remote server
+     * @return an array of <code>FTPFile</code> objects
+     */
+    protected FTPFile[] listRemoteFiles() throws IOException{
+        String path = ftp.printWorkingDirectory();
+
+        FTPFile[] files =  ftp.listFiles(path, new FTPFileFilter() {
+            @Override
+            public boolean accept(FTPFile ftpFile) {
+                return !ftpFile.isDirectory();
+            }
+        });
+
+        return files;
+    }
+
+    /**
+     * Converts names of <code>FTPFile</code> objects to strings
+     * @param files array of <code>FTPFiles</code> objects
+     * @return <code>ArrayList</code> of names in <code>String</code> format
+     */
+    protected ArrayList<String> fileDirectoryListStrings(FTPFile[] files){
+       ArrayList<String> names = null;
+
+       for(FTPFile file:files)
+           names.add(file.getName());
+
+       return names;
+    }
+
+    /**
+     * Returns <code>String</code> of the current working directory
+     * @return current working directory
+     * @throws IOException
+     */
+    protected String printWorkingDirectory() throws IOException{
+        return ftp.printWorkingDirectory();
+    }
+
+    /**
+     * Changes the working directory...
+     * @param path ...to the given file path
+     * @return true if the path change was successful, false otherwise
+     */
+    protected boolean changeDirectory(String path){
```

Comment:
changeDirectory returns false if the user tries to change the cwd to their cwd. Add logic to check for this condition, and return before sending any requests if true.